### PR TITLE
Added JetBrains Support

### LINKED ISSUE #9: Add support for JetBrians IDEs installed through Toolbox
### Acknowledgements

- [X] I have read the list of [currently supported code editors and IDEs](https://github.com/realmazharhussain/nautilus-code#currently-supported) and I don't see it listed there.

### Editor/IDE website (Optional)

https://www.jetbrains.com/toolbox-app/download/other.html

The IDEs installed are usually found in `~/.local/share/JetBrains/Toolbox/apps`.

### PR DESCRIPTION
**Pull request by BVengo**
Added native and flatpak support for Jetbrains IDEs.

All of these apply to installations via Jetbrains Toolbox as well, since it creates links for the native commands (resolves #9).

Every new IDE was installed and tested (both Native and Flatpak)  to confirm that they open projects as expected and don't require additional arguments. Language support will still need updating with all the new IDEs, however I also noticed that many of the translation files already had incorrect line numbers in the comments.

List of Jetbrains IDEs:
- Android Studio
- Aqua
- CLion
- DataGrip
- DataSpell
- Fleet
- GoLand
- IntelliJ
- MPS
- PhpStorm (already supported)
- PyCharm
- Rider
- RubyMine
- WebStorm

### CODE REVIEW

**Comment by realmazharhussain** [comment]
Some basic MR (Merge/Pull Request) ethics include;

* Don't do anything more than what is described in the title of the MR
* Use the same coding style as already being used in the files you are modifying.

The docstrings you have added and the changes you have made in commit 2dfae6de6183ffd4f27e4aa046ad8fd3211ebda2 violate both of these ethics. They are good changes but do not belong in this MR.

Please, fix your MR. Thanks.

**Comment by BVengo** [comment]
My mistake, thanks for the info! I've removed all extra docstrings and refactoring.

File: NautilusCode/program_list.py [hunk 1]
```diff
@@ -34,9 +34,97 @@
...
+                 Flatpak('com.jetbrains.IntelliJ-IDEA-Ultimate'))
+
+progs += Program('mps', _('MPS'),
+                 Native('mps'))
+
 progs += Program('phpstorm', _("PhpStorm"),
-                 Flatpak("com.jetbrains.PhpStorm"),
-                 Native("phpstorm"))
+                 Native("phpstorm"),
+                 Flatpak("com.jetbrains.PhpStorm"))
```
Comment (realmazharhussain) [inline NautilusCode/program_list.py:94]: This is an unrelated/unecessary change and should not be part of this MR.

**Review by realmazharhussain — CHANGES_REQUESTED** [review]

File: NautilusCode/program_list.py (hunk 1 above)
Comment (BVengo) [reply NautilusCode/program_list.py:94]: Is this not a part of adding support for all JetBrains IDEs by maintaining consistency between how they are coded (including it's alphabetical position in the code)?

I'll revert the change, however I didn't consider that to be a refactor but instead an implementation that includes all products instead of just the one.

**Comment by realmazharhussain** [comment]
Thank you for all this work.

Now, one final change remains. There should only be two commits:

* Add JetBrains IDEs support
* README.md: Add Jetbrains IDEs

Every other change should be squashed into these commits.

File: NautilusCode/program_list.py (hunk 1 above)
Comment (realmazharhussain) [reply NautilusCode/program_list.py]: Well, nautilus-code already supports PhpStorm.

So, this change does not fit commit "**Add** JetBrains support".

A better way to do this would be to sort the whole list in a single commit outside this MR.

**Review by realmazharhussain — APPROVED** [review]

**Comment by BVengo** [comment]
Thanks for your patience - clearly I don't make pull requests often!

Changes are all squashed, and it should be good to go.

**Comment by realmazharhussain** [comment]
Also, I revisited issue #9 and this does not resolve #9. The author of #9 specifically mentions JetBrains Toolbox.

So, please, remove the mention of #9 from the MR description. Thanks.

**Comment by BVengo** [comment]
All of these apply to installations via Jetbrains Toolbox as well, since it
creates links for the native commands.

I'll update the description to specify that.

**Comment by realmazharhussain** [comment]
> it creates links for the native commands

Really! I didn't know that. That's great.

**Comment by BVengo** [comment]
Yep! Very useful, and I checked them all to make sure they all did it by default.

**Comment by realmazharhussain** [comment]
Hi, git works in fricking mysterious ways. Can you re-open the MR after re-pushing to your main branch? Thanks.

**Comment by BVengo** [comment]
I noticed that, and it's confused me to no end.

Will re-open in 15 once I get home again.
